src/commands/giveaway.rs

```diff
@@ -0,0 +1,650 @@
+                debug!(
+                    "Showing next {} giveaways for user {}",
+                    embeds.len(),
+                    component.user.id.0
+                );
+                component
+                    .create_interaction_response(&ctx.http, |r| {
+                        r.kind(InteractionResponseType::UpdateMessage);
+                        r.interaction_response_data(|d| {
+                            d.set_components(components);
+                            d.embeds(embeds)
+                        })
+                    })
+                    .await
+                    .unwrap();
+                offsets.insert(component.user.id.0, offset);
+            }
+            _ => debug!("Unknown interaction: {}", component.data.custom_id),
+        }
+    }
+}
+
+pub async fn handle_interaction(ctx: &Context, interaction: ApplicationCommandInteraction) {
+    let db = ctx.get_db().await;
+
+    ensure_offset_map(&ctx).await;
+    let mut data = ctx.data.write().await;
+    let offset_map = data.get_mut::<ListOffset>().unwrap();
+
+    let option = interaction
+        .data
+        .options
+        .first()
+        .expect("Giveaway subcommand missing");
+    let sub_options = option.options.clone();
+
+    match option.name.as_str() {
+        "start" => {
+            let channel = sub_options
+                .by_name("channel")
+                .expect("Missing channel option")
+                .to_channel()
+                .expect("Invalid channel option");
+            let duration = sub_options
+                .by_name("duration")
+                .map_or(DEFAULT_DURATION, |x| x.to_i64().unwrap_or(DEFAULT_DURATION));
+            let winners = sub_options
+                .by_name("winners")
+                .map_or(DEFAULT_WINNERS, |x| x.to_i64().unwrap_or(DEFAULT_WINNERS));
+            let prize = sub_options
+                .by_name("prize")
+                .map_or(DEFAULT_PRIZE.to_string(), |x| {
+                    x.to_string().unwrap_or(DEFAULT_PRIZE.to_string())
+                });
+
+            if winners < 1 || duration < 1 {
+                interaction
+                    .create_interaction_response(&ctx.http, |r| {
+                        r.interaction_response_data(|d| {
+                            d.content(
+                                "Duration and winners must be positive integers or left empty.",
+                            );
+                            d.flags(InteractionApplicationCommandCallbackDataFlags::EPHEMERAL)
+                        })
+                    })
+                    .await
+                    .unwrap();
+            } else {
+                let now = chrono::Utc::now();
+                let end = now + chrono::Duration::seconds(duration);
+                let mut message = channel
+                    .id
+                    .send_message(&ctx.http, |c| {
+                        c.content("@everyone");
```

Comment:
The `@everyone` mention should probably be optional / to be able to be disabled by either the creator of the giveaway, or in the `.env` file.